Updated node roles in docs (#49360)
kubernetes/website: content/en/docs/concepts/architecture/nodes.md

You can modify Node objects regardless of the setting of `--register-node`.
For example, you can set labels on an existing Node or mark it unschedulable.

> You can set optional node role(s) for nodes by adding one or more `node-role.kubernetes.io/<role>: <role>` labels to the node where characters of `<role>` are limited by the [syntax](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) rules for labels.

Reviewer: ```suggestion
You can mark optional node role(s) for nodes by adding one or more `node-role.kubernetes.io/<role>: <role>` labels to the node where characters of `<role>`
are limited by the [syntax](/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) rules for labels.
```

You can modify Node objects regardless of the setting of `--register-node`.
For example, you can set labels on an existing Node or mark it unschedulable.

You can set optional node role(s) for nodes by adding one or more `node-role.kubernetes.io/<role>: <role>` labels to the node where characters of `<role>` 

> are limited by the [syntax](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) rules for labels.

Reviewer: ```suggestion
are limited by the [syntax](/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) rules for labels.
```

> Kubernetes ignores the label value for node roles;by convention, you can set it to the same string you used for the node role in label key.

Reviewer 2: ```suggestion
Kubernetes ignores the label value for node roles; by convention, you can set it to the same string you used for the node role in the label key.
```

Author: I have addressed this in the new commit. Thanks for pointing this out.

Reviewer 2: You are welcome

You can modify Node objects regardless of the setting of `--register-node`.
For example, you can set labels on an existing Node or mark it unschedulable.

> You can set optional node role for nodes by adding `node-role.kubernetes.io/<role>=` label to the node where characters of `<role>` are limited by the [syntax](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) rules of labels.

Reviewer: Does this look right?
```suggestion
You can set optional node role(s) for nodes by adding one or more `node-role.kubernetes.io/<role>: <role>` labels to the node
where the characters of `<role>` are limited by the
[syntax](/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) rules for labels.

Kubernetes ignores the label value for node roles; by convention, you set it to the same string you used
for the node role in the label key.
```

Author: Thanks for your suggestion! Everything looks great. There is one thing I wanted to mention:
`kubectl label nodes <node_name> node-role.kubernetes.io/<role>=` command adds a role to the node. If I use `kubectl label nodes <node_name> node-role.kubernetes.io/<role>:<role>`, then it gives the following error `error: at least one label update is required`.
So, I am not sure if `node-role.kubernetes.io/<role>:<role>` can be used.

Reviewer: In a manifest (YAML), you use a `:`. On the command line, `=`. In this file we write the labels and annotations as if they are part of a YAML manifest.

Author: Thanks for clearing this. I have made the changes.
